shared-bindings/_bleio/Adapter.c

```diff
@@ -71,6 +71,59 @@
 //|         ...
 //|
 
+//|     def hci_uart_init(self, *, uart: busio.UART, rts: digitalio.DigitalInOut, cts: digitalio.DigitalInOut, baudrate: int = 115200, buffer_size: int = 256) -> None:
```

Comment:
I originally did it this way in rough imitation of `displayio.DISPLAY`. There is no explicit constructor for the default display, and there wouldn't be for the singleton adapter either. Unlike DISPLA`, there does need to be some initialization done to use the singleton adapter. The idea was that it would live across VM's, like the DISPLAY.

It is possible to make the Adapter not be a singleton, but it requires passing it around many more places than it is currently. Right now it's referenced as a global in a number of places. The static declarations in `att.c` and `hci.c` would also need to be merged into the Adapter instance. I started to do this but it's quite a bit of work that could be deferred for now.

Reply:
Changes: I made Adapter have a constructor that takes the uart, rts, and cts args. You can actually only call it once, before it complains "Too many adapters". To set `_bleio.adapter`, you call `_bleio.set_adapter()`. It can't be used as a settable property due to limitations in how module attributes are treated.

In an HCI build, the attribute is set back to `None` on each soft restart. In a non-HCI build (e.g. nRF), the attribute is always set and is read-only.